components/core/src/clp/ffi/ir_stream/ir_unit_deserialization_methods.hpp

```diff
@@ -63,6 +65,8 @@ namespace clp::ffi::ir_stream {
  * - std::errc::protocol_error if the IR stream is corrupted.
  * - std::errc::protocol_not_supported if the IR stream contains an unsupported metadata format
  *   or uses an unsupported version.
+ * - std::errc::protocol_not_supported if the IR stream contains auto-generated keys.
+ *   TODO: remove this once auto-generated keys are fully supported.
```

Comment:
```suggestion
 * - std::errc::protocol_not_supported if the IR stream contains an unsupported metadata format,
 *   uses an unsupported version, or if the IR stream contains auto-generated keys (TODO: Remove
 *   this case when auto-generated keys are fully supported).
```

Or you could use `Forwards `deserialize_schema`'s return values.`?

Reply:
Make sense. I will use `forward` version